front/cypress/e2e/admin/pages_and_menu/create_update_and_view_custom_page.cy.ts

```diff
@@ -0,0 +1,154 @@
+    cy.intercept('GET', '**/static_pages/**').as('getCustomPage');
+
+    const pageName = randomString();
+    const headerContent = randomString();
+    const subheaderContent = randomString();
+    const ctaContent = randomString();
+    const topInfoContent = randomString();
+
+    // go to custom page creation form
+    cy.get('#create-custom-page').click();
+
+    // type title in each language
+    cy.get('.e2e-localeswitcher').each((button) => {
+      cy.wrap(button).click();
+      cy.get('#title_multiloc').type(pageName);
+    });
+
+    // submit
+    cy.get('[data-cy="e2e-submit-custom-page"]').click();
+
+    // click the page content tab
+    cy.contains('Page content').click();
+
+    const sectionNames = [
+      'banner_enabled',
+      'top_info_section_enabled',
+      'files_section_enabled',
+    ];
+    // toggle top info section and wait for requests to complete
+    sectionNames.forEach((sectionName) => {
+      // click the toggle
+      cy.get(`[data-cy="e2e-admin-section-toggle-${sectionName}"]`).click();
+      // wait for the call to complete
+      cy.wait('@updateCustomPage');
+      // wait for the toggle to become enabled in the UI
+      cy.get(`[data-cy="e2e-admin-section-toggle-${sectionName}"]`)
+        .find('i')
+        .should('have.class', 'enabled');
+    });
+
+    // go to the hero banner edit
+    cy.get('[data-cy="e2e-admin-edit-button-banner_enabled"]').click();
+
+    // focus the header dropzone and drop an image onto it
+    cy.get('#header-dropzone').attachFile('icon.png', {
+      subjectType: 'drag-n-drop',
+    });
+
+    // update the header text
+    cy.get('[data-cy="e2e-signed-out-header-section"]').scrollIntoView();
+    cy.get('[data-cy="e2e-signed-out-header-section"]')
+      .find('.e2e-localeswitcher')
+      .each((button) => {
+        cy.wrap(button).click();
+        cy.get('[data-cy="e2e-signed-out-header-section"]')
+          .find('input')
+          .type(headerContent);
+      });
+
+    // update the subheader text
+    cy.get('[data-cy="e2e-signed-out-subheader-section"]').scrollIntoView();
+    cy.get('[data-cy="e2e-signed-out-subheader-section"]')
+      .find('.e2e-localeswitcher')
+      .each((button) => {
+        cy.wrap(button).click();
+        cy.get('[data-cy="e2e-signed-out-subheader-section"]')
+          .find('input')
+          .type(subheaderContent);
+      });
+
+    // enable custom button
+    cy.get('[data-cy="e2e-cta-settings-custom-customized_button"]').click();
+
+    cy.get('[data-cy="e2e-cta-settings-custom-customized_button"]')
+      .find('.e2e-localeswitcher')
+      .each((button) => {
+        cy.wrap(button).click();
+        cy.get('[data-testid="inputMultilocLocaleSwitcher"]')
+          .find('input')
+          .type(ctaContent);
+      });
+
+    // type the url
+    cy.get('#buttonConfigInput').type('https://www.coolwebsite.biz');
+
+    // submit form
+    cy.get('.e2e-submit-wrapper-button').click();
```

Comment:
Shouldn't we check it's saved successfully?

Reply:
Though it's not required (anyway the applied data is checked on the page for citizens), it may help:
1. Detect issues with the toast or the form (data is saved, but it's not shown to the admin or an error is shown)
2. Detect issues with saving data at this stage, not waiting till the end of the test. So, it will be immediately clear that the problem is related to the form and not to the presentation of a page for citizens.

Not a big deal though.

Reply:
Yeah, good call, on re-reading it seems like there could be a condition where we navigate off the page before the call has completed. I'm not sure if that would make the call fail, but it also does make more sense to wait for the success toast either way. I'll add it